bin/run_stager_in_ec2.sh

```diff
@@ -0,0 +1,135 @@
+    new_state=`echo ${terminate_instances_response_json} | jq .TerminatingInstances[0].CurrentState.Name`
+    echo "New state is ${new_state}"
+    aws ec2 wait instance-terminated --instance-ids ${instance_id}
+}
+
+install_and_setup_docker_on_ec2_instance(){
+    echo -e "\n########## Installing Docker ##########\n"
+    ssh_ec2 <<-EOF
+        set -x
+		if [ \`apt-key list | grep Docker | wc -l\` -ne 1 ] ; then
+		    echo "Adding docker apt repository"
+			curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+		fi
+		grep -nq docker /etc/apt/sources.list || \
+		sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu \$(lsb_release -cs) stable"
+		(dpkg -l | grep -q docker) || sudo apt-get update && sudo apt-get install -y docker-ce htop
+		egrep -q "docker:.*ubuntu" /etc/group || sudo usermod -aG docker \${USER}
+	EOF
+}
+
+setup_git_repo(){
+    echo -e "\n########## Setting up Git Repo ########## \n"
+    ssh_ec2 <<-EOF
+        set -x
+        mkdir -p ~/.ssh
+        grep -q github.com ~/.ssh/config || echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
+		[ -d data-bundle-examples ] || git clone --recursive git@github.com:humancellatlas/data-bundle-examples.git
+		cd data-bundle-examples
+		git checkout spierson-stager
+	EOF
+}
+
+build_docker_image(){
+    echo -e "\n########## Building Docker Image ##########\n"
+    ssh_ec2 <<-EOF
+        set -x
+        cd data-bundle-examples
+        cat > Dockerfile <<EODOCKERFILE
+FROM python:3.6
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+  vim
+WORKDIR /code/data-bundle-examples
+ADD requirements.txt /code/data-bundle-examples
+RUN pip install -r requirements.txt
+ENV AWS_ACCESS_KEY_ID=${aws_access_key_id}
+ENV AWS_SECRET_ACCESS_KEY=${aws_secret_access_key}
+CMD /code/data-bundle-examples/bin/stager.py -j10 --log /code/data-bundle-examples/stager.log
+EODOCKERFILE
+	docker build . --tag hca-stager
+	EOF
+}
+
+run_stager(){
+    echo -e "\n########## Running Stager ##########\n"
+    ssh_ec2 <<-EOF
+        set -x
+        cd data-bundle-examples
+        [ -f import/10x/t_4k/bundles/bundle1/assay.json ] || tar xf import/import.tgz
+        # docker run --rm --name hca-stager -v \${HOME}/data-bundle-examples:/code/data-bundle-examples hca-stager
+        docker run --name hca-stager -v \${HOME}/data-bundle-examples:/code/data-bundle-examples hca-stager
+	EOF
+}
+
+ssh_ec2(){
+    ssh -oStrictHostKeyChecking=no ubuntu@${public_dns} $*
+}
+
+abort(){
+    terminate_ec2_instance
+    exit 1
+}
+
+grab_aws_creds
+grab_keypair_name
+echo -e "Starting at `date`"
+boot_ec2_instance
+trap abort SIGINT
+wait_for_ec2_boot_to_complete
+install_and_setup_docker_on_ec2_instance
+setup_git_repo
+build_docker_image
+run_stager
+terminate_ec2_instance
+echo -e "Finished at `date`"
+exit 0
```

Comment:
I wish you'd told me about what you were doing. I'd recommend using https://github.com/kislyuk/aegea for this